data/model/oci/tag.py

```diff
@@ -432,18 +432,19 @@ def delete_tag(repository_id, tag_name):
     if tag is None:
         return None
 
-    return _delete_tag(tag, get_epoch_timestamp_ms())
+    return _delete_tag(tag, get_epoch_timestamp_ms(), force)
 
 
-def _delete_tag(tag, now_ms):
+def _delete_tag(tag, now_ms, force=False):
     """
     Deletes the given tag by marking it as expired.
     """
-    now_ts = int(now_ms // 1000)
-
     with db_transaction():
+        # Force set's lifetime_end_ms outside of any time machine policy and hidden
+        # prevents the tag from appearing in tag history for restoration
+        params = {"lifetime_end_ms": 0, "hidden": True} if force else {"lifetime_end_ms": now_ms}
```

Comment:
Any reason to specifically hide the tag? The quota won't be updated until GC runs on that repo anyways.

Reply:
Also it looks like there's a unique id built off the name and lifetime_end_ms so i think i'll have to update the `lifetime_end_ms` to just outside the time machine window.